specification/dartLangSpec.tex

```diff
@@ -11641,6 +11765,246 @@ \subsection{Assignable Expressions}
 Evaluation of an assignable expression of the form \code{\SUPER{}[$e_2$]} is equivalent to evaluation of the method invocation \code{\SUPER{}.[]($e_2$)}.
 
 
+\subsection{Lexical Lookup}
+\LMLabel{lexicalLookup}
+
+\LMHash{}%
+This section specifies how to look up a name
+based on the enclosing lexical scopes.
+This is known as a \Index{lexical lookup}.
+When \id{} is an identifier, it may look up a name $n$ of the form \id{}
+as well as of the form \code{\id=}.
+
+\LMHash{}%
+A lexical lookup does not succeed or fail.
+It yields a result which is
+a declaration, an import prefix, or a member signature,
+unless a compile-time error occurs during the lexical lookup.
+
+\commentary{%
+A lexical lookup differs from a lookup of an instance member
+(\ref{lookup})
+because that operation searches through a sequence of superclasses,
+whereas a lexical lookup searches through a sequence of enclosing scopes.
+A lexical lookup differs from a straightforward lookup in the enclosing scopes
+because the lexical lookup ``bundles'' getters and setters,
+as detailed below.%
+}
+
+\LMHash{}%
+Consider the situation where a name $n$ has basename \id{}
+(\ref{classMemberConflicts}),
+an identifier \id{} occurs at a location $\ell$,
+and a lexical lookup of $n$ is performed from $\ell$.
+
+\commentary{%
+We need to specify a name and a location from where a lookup is performed.
```

Comment:
Drop "need". We *do*. 
We probably wouldn't do it unless we needed to, but you don't need to defend the choice :).

Reply:
This is about the _reason_ for mentioning $\ell$ in the definition (it's a new concept in the specification). The spec never talked about 'locations' before, that was always just implicit in the fact that the spec generally talks about a snippet of code which has a location because it's known to be part of the program (or it's a desugaring which is, implicitly, considered to be in the same location as the original).

I rewrote the whole paragraph, avoiding `need`, and the result is clearer.